tests/integration/aof-multi-part.tcl

```diff
@@ -0,0 +1,1065 @@
+set aof_base2_filepath "$server_path/$aof_dirname/${aof_basename}.2$::base_aof_sufix$::aof_format_suffix"
+set aof_incr1_filepath "$server_path/$aof_dirname/${aof_basename}.1$::incr_aof_sufix$::aof_format_suffix"
+set aof_incr2_filepath "$server_path/$aof_dirname/${aof_basename}.2$::incr_aof_sufix$::aof_format_suffix"
+set aof_incr3_filepath "$server_path/$aof_dirname/${aof_basename}.3$::incr_aof_sufix$::aof_format_suffix"
+set aof_manifest_filepath "$server_path/$aof_dirname/${aof_basename}$::manifest_suffix"
+set aof_old_version_filepath "$server_path/$aof_basename"
+set aof_old_version_newpath "$aof_dirpath/$aof_basename"
+set aof_old_version_filepath2 "$server_path/${aof_basename}2"
+set aof_manifest_filepath2 "$server_path/$aof_dirname/${aof_basename}2$::manifest_suffix"
+
+tags {"external:skip"} {
+    
+    # Test Part 1
+    #
+    # In order to test the loading logic of redis under different combinations of manifest and AOF.
+    # We will manually construct the manifest file and AOF, and then start redis to verify whether 
+    # the redis behavior is as expected.
+
+    # Tests1: Multi Part AOF can't load data when some file missing
+    create_aof_dir $aof_dirpath
+    create_aof $aof_base1_filepath {
+        append_to_aof [formatCommand set k1 v1]
+    }
+
+    create_aof $aof_incr2_filepath {
+        append_to_aof [formatCommand set k2 v2]
+    }
+
+    create_aof_manifest $aof_manifest_filepath {
+        append_to_manifest "file appendonly.aof.1.base.aof seq 1 type b\n"
+        append_to_manifest "file appendonly.aof.1.incr.aof seq 1 type i\n"
+        append_to_manifest "file appendonly.aof.2.incr.aof seq 2 type i\n"
+    }
+
+    start_server_aof [list dir $server_path] {
+        test "Multi Part AOF can't load data when some file missing" {
+            wait_for_condition 100 50 {
+                ! [is_alive $srv]
+            } else {
+                fail "AOF loading didn't fail"
+            }
+
+            assert_equal 1 [count_message_lines $server_path/stdout "doesn't exist"]
+        }
+    }
+
+    clean_aof_persistence $aof_dirpath
+
+    # Tests2: Multi Part AOF can't load data when the sequence not increase monotonically
+    create_aof_dir $aof_dirpath
+    create_aof $aof_incr1_filepath {
+        append_to_aof [formatCommand set k1 v1]
+    }
+
+    create_aof $aof_incr2_filepath {
+        append_to_aof [formatCommand set k2 v2]
+    }
+
+    create_aof_manifest $aof_manifest_filepath {
+        append_to_manifest "file appendonly.aof.2.incr.aof seq 2 type i\n"
+        append_to_manifest "file appendonly.aof.1.incr.aof seq 1 type i\n"
+    }
+
+    start_server_aof [list dir $server_path] {
+        test "Multi Part AOF can't load data when the sequence not increase monotonically" {
+            wait_for_condition 100 50 {
+                ! [is_alive $srv]
+            } else {
+                fail "AOF loading didn't fail"
+            }
+        }
+    }
+
+    clean_aof_persistence $aof_dirpath
```

Comment:
maybe it'll be better if each of these tests is in a test scope.
like so:
```tcl
test {name / description} {
    create_aof_dir ...
    create_aof_manifest ....

    start_server_aof [list dir $server_path] {
        wait_for_condition 100 50 {
            ! [is_alive $srv]
        } else {
            fail "AOF loading didn't fail"
        }
        assert_something ...
    }

    clean_aof_persistence $aof_dirpath
}
```
i.e. the server is started inside the test and not vice versa.
p.s. this way if the test has some skip tag, the server setup and creation is skipped too

Reply:
OK